### PR TITLE
zh-translation: /docs/tasks/configure-pod-container/configure-livenes…

### PR DESCRIPTION
**Pull request by helight**
zh-translation: /docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md

ref: https://github.com/k8smeetup/website-tasks/issues/3651

### CODE REVIEW

File: content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md [hunk 1]
```diff
@@ -0,0 +1,530 @@
+---
+title: 配置存活，就绪和启动探测器
```
Comment (tengqm) [inline content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md]:
```suggestion
title: 配置存活、就绪和启动探测器
```

File: content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md [hunk 2]
```diff
@@ -0,0 +1,530 @@
...
+<!--
+This page shows how to configure liveness, readiness and startup probes for Containers.
+
+The [kubelet](/docs/admin/kubelet/) uses liveness probes to know when to
+restart a Container. For example, liveness probes could catch a deadlock,
+where an application is running, but unable to make progress. Restarting a
+Container in such a state can help to make the application more available
+despite bugs.
+-->
+这篇文章介绍如何给容器配置存活，就绪和启动探测器。
```
Comment (tengqm) [inline content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md]:
```suggestion
这篇文章介绍如何给容器配置存活、就绪和启动探测器。
```

File: content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md [hunk 3]
```diff
@@ -0,0 +1,530 @@
...
+
+The kubelet uses startup probes to know when a Container application has started.
+If such a probe is configured, it disables liveness and readiness checks until
+it succeeds, making sure those probes don't interfere with the application startup.
+This can be used to adopt liveness checks on slow starting containers, avoiding them
+getting killed by the kubelet before they are up and running.
+-->
+kubelet 使用就绪探测器可以知道容器什么时候准备好了并可以开始接受请求流量， 当一个 Pod 内的所有容器都准备好了，才能把这个 Pod 看作就绪了。这种信号的一个用途就是控制哪个 Pod 作为 Service 的后端。在 Pod 还没有准备好的时候，会从 Service 的负载均衡器中被剔除的。
+
+kubelet 使用启动探测器可以知道应用程序容器什么时候启动了。如果这样的探测器可以配置，就可以控制容器在启动成功后再进行存活性和就绪检查，确保这些存活、就绪探测器不会影响应用程序的启动。这可以用于对慢启动容器进行存活性检测，避免它们在启动运行之前就被杀掉。
```
Comment (tengqm) [inline content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md]:
```suggestion
kubelet 使用启动探测器可以知道应用程序容器什么时候启动了。如果配置了这类探测器，就可以控制容器在启动成功后再进行存活性和就绪检查，确保这些存活、就绪探测器不会影响应用程序的启动。这可以用于对慢启动容器进行存活性检测，避免它们在启动运行之前就被杀掉。
```

File: content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md [hunk 4]
```diff
@@ -0,0 +1,530 @@
...
+创建 Pod：
+
+```shell
+kubectl apply -f https://k8s.io/examples/pods/probe/exec-liveness.yaml
+```
+
+<!--
+Within 30 seconds, view the Pod events:
+-->
+在 30 秒内，看 Pod 的事件：
```
Comment (tengqm) [inline content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md]:
```suggestion
在 30 秒内，查看 Pod 的事件：
```

File: content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md [hunk 5]
```diff
@@ -0,0 +1,530 @@
...
+```
+
+<!--
+## Define a liveness HTTP request
+
+Another kind of liveness probe uses an HTTP GET request. Here is the configuration
+file for a Pod that runs a container based on the `k8s.gcr.io/liveness`
+image.
+-->
+## 定义一个存活的 HTTP 请求接口 {#define-a-liveness-HTTP-request}
```
Comment (tengqm) [inline content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md]:
```suggestion
## 定义一个存活态 HTTP 请求接口 {#define-a-liveness-HTTP-request}
```

File: content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md [hunk 6]
```diff
@@ -0,0 +1,530 @@
...
+
+另外一种类型的存活探测方式是使用 HTTP GET 请求，下面是一个 Pod 的配置文件，其中运行一个基于 `k8s.gcr.io/liveness` 镜像的容器。
```
Comment (tengqm) [inline content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md]:
```suggestion
另外一种类型的存活探测方式是使用 HTTP GET 请求。下面是一个 Pod 的配置文件，其中运行一个基于 `k8s.gcr.io/liveness` 镜像的容器。
```

File: content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md [hunk 7]
```diff
@@ -0,0 +1,530 @@
...
+
+<!--
+Thanks to the startup probe, the application will have a maximum of 5 minutes
+(30 * 10 = 300s) to finish its startup.
+Once the startup probe has succeeded once, the liveness probe takes over to
+provide a fast response to container deadlocks.
+If the startup probe never succeeds, the container is killed after 300s and
+subject to the pod's `restartPolicy`.
+-->
+辛苦有启动探测，应用程序将会有最多 5 分钟(30 * 10 = 300s) 的时间来完成它的启动。
```
Comment (tengqm) [inline content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md]:
```suggestion
幸亏有启动探测，应用程序将会有最多 5 分钟(30 * 10 = 300s) 的时间来完成它的启动。
```

File: content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md [hunk 8]
```diff
@@ -0,0 +1,530 @@
...
+一旦启动探测成功一次，存活探测任务就会接管对容器的探测，对容器死锁可以快速响应。
+如果启动探测一直没有成功，容器会在 300 秒后被杀死，并且设置 Pod 状态为 `restartPolicy`。
```
Comment (tengqm) [inline content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md]:
```suggestion
如果启动探测一直没有成功，容器会在 300 秒后被杀死，并且根据`restartPolicy` 来设置 Pod 状态。
```

File: content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md [hunk 9]
```diff
@@ -0,0 +1,530 @@
...
+files during startup, or depend on external services after startup.
+In such cases, you don't want to kill the application,
+but you don’t want to send it requests either. Kubernetes provides
+readiness probes to detect and mitigate these situations. A pod with containers
+reporting that they are not ready does not receive traffic through Kubernetes
+Services.
+-->
+## 定义就绪探测器 {#define-readiness-probes}
+
+有时候，应用程序会暂时性的不能提供通信服务。例如，应用程序在启动时可能需要加载很大的数据或配置文件，或是启动后要依赖等待外部服务。在这种情况下，即不想去杀死应用程序，也不想给它发送请求。Kubernetes 提供了就绪探测器来发现并缓解这些情况。容器所在 Pod 上报还未就绪的信息，并且不接受通过 Kubernetes Service 的流量。
```
Comment (tengqm) [inline content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md]:
```suggestion
有时候，应用程序会暂时性的不能提供通信服务。例如，应用程序在启动时可能需要加载很大的数据或配置文件，或是启动后要依赖等待外部服务。在这种情况下，既不想杀死应用程序，也不想给它发送请求。Kubernetes 提供了就绪探测器来发现并处理这些情况。若容器所在 Pod 上报还未就绪的信息，则 Pod 不接受通过 Kubernetes Service 的流量。
```

File: content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md [hunk 10]
```diff
@@ -0,0 +1,530 @@
...
+
+{{< note >}}
+<!--
+Readiness probes runs on the container during its whole lifecycle.
+-->
+就绪探测器运行在容器的整个生命周期。
```
Comment (tengqm) [inline content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md]:
```suggestion
就绪探测器在容器的整个生命周期中保持运行状态。
```

File: content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md [hunk 11]
```diff
@@ -0,0 +1,530 @@
...
+<!--
+## Configure Probes
+-->
+## 配置探测器 {#configure-probes}
+
+{{< comment >}}
+<!--
+Eventually, some of this section could be moved to a concept topic.
+-->
+最后，本节的一些内容可以放到概念这个主题里。
```
Comment (tengqm) [inline content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md]:
```suggestion
最后，本节的一些内容可以放到某个概念主题里。
```

File: content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md (hunk 1 above)
Comment (helight) [reply content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md]: accept, thanks.

File: content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md (hunk 2 above)
Comment (helight) [reply content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md]: accept, thanks.

File: content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md (hunk 3 above)
Comment (helight) [reply content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md]: accept, thanks.

File: content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md (hunk 4 above)
Comment (helight) [reply content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md]: accept, thanks.

File: content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md (hunk 5 above)
Comment (helight) [reply content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md]: accept, thanks.

File: content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md (hunk 6 above)
Comment (helight) [reply content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md]: accept, thanks.

File: content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md (hunk 7 above)
Comment (helight) [reply content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md]: accept, thanks.

File: content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md (hunk 8 above)
Comment (helight) [reply content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md]: accept, thanks.

File: content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md (hunk 9 above)
Comment (helight) [reply content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md]: accept, thanks.

File: content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md (hunk 10 above)
Comment (helight) [reply content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md]: accept, thanks.

File: content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md (hunk 11 above)
Comment (helight) [reply content/zh/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes.md]: accept, thanks.

**Pull request merged**